### DISCUSSION TITLE
Feature to add creation of project-specific BibTeX files from manuscript sources

### DISCUSSION THREAD

**Discussion by CodePurble** [papis/papis#408] [General]
I have often come across the need to create project/manuscript-specific BibTeX files from my global database of bibliographic entries. I wrote a [script to do just that](https://github.com/CodePurble/bibextract). I was wondering if such a thing can be done using `papis`. If not, if it is something that aligns with the goals of `papis`, I am ready to integrate what I have done in the script into `papis`.

I would appreciate any comments and suggestions on this.
Thanks

**Comment by jghauser** [comment]
When doing `papis export` you can specify a query that defines which documents should be exported. For instance, I have 'meta' tags that specify what project an item belongs to. With that I can then do:
```
papis export --format bibtex --out project1.bib 'tags:meta_project1'
```
You can find [more information](https://papis.readthedocs.io/en/latest/commands.html#module-papis.commands.export) in the docs.

Does that help?

EDIT: I just had a look at your script and realised that you're parsing your document for the relevant reference keys. Sorry to have missed that! I don't think this is possible with papis at the moment, and I'm not sure whether such a feature wouldn't take papis a little too far afield from its core functionality (but maybe @alejandrogallo and @alexfikl can chime in). It should certainly be easy to adapt your script to work with papis. Maybe this could be an additional script similarly to the ones we already have in `examples/scripts/`. Might be nice to expand and document these user scripts.

**Reply by CodePurble** [reply]
Thanks for the tip! In the meantime, I'll try to write-up an example script that uses `papis`.

**Comment by alejandrogallo** [comment]
there is a functionality I implemented some time ago and I use for my projects, which is using the bibtex command, you can then read and write and curate your project's bibfile.

I tried to document  this workflow here 

https://papis.readthedocs.io/en/latest/commands.html#local-configuration-file

but feel free to suggest improvements or ask away !
I hope this helps a bit
